### ISSUE TITLE
How can I get logits from transformer model ?

### ISSUE THREAD

**Issue by efeiefei** [tensorflow/tensor2tensor#512]
Actually, I want to get the likelihood score for a specific <`source seq`, `target seq`> pair. 
I will use it to rerank decode results with decode score. 
For example, translate sequence `s` to 10 results with 10 score, then compute <`res1`, `s`> ... <`res10`, `s`>  likelihood score use another independent reverse model . Finally, use linear model to rerank to select the best one.

Thx!

**Comment by efeiefei** [comment]
Can some on help me?

I have tried to pick out the logits/loss operation. But I can't select a pure model directly and determine input features format, as Tensor2Tensor project is some complicated.

**Comment by yhg0112** [comment]
i understand it is complicated.
i suppose you have to check the modality part. 
Modifying
https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/modality.py or 
https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/layers/modalities.py
, you can do what you want.

i saw my co-workers in my lab who aren't too familiar with tf.experiments use only the model code to experiment their own model. you should also try that way.

**Comment by efeiefei** [comment]
@yhg0112 
Very grateful to you, I will try that way.

**Comment by efeiefei** [comment]
I have solved this problem, with the example `utils/trainer_utils_test.py`.
The example is in t2t `version 1.3.2`, I changed a little so I can use it in `version 1.4.1`

**Comment by zqma2** [comment]
Recently, I met the same problem. Can you share the solution to this problem？
Thx
